proto/plugin.proto

```diff
@@ -739,21 +739,24 @@ message Deploy {
 
 // A platform resource that an operation (release/deployment) has created, depends on, or manages.
 message DeclaredResource {
+  // The id of the declared resource. Automatically generated
+  string id = 1;
```

Comment:
This was missing from the original RRC - we need an ID here so that Resources can tie pack to a declared_resource through this ref:

https://github.com/hashicorp/waypoint/blob/c8e5b0ba25a4f6e9f804499495f288e66b0b5c21/internal/server/proto/server.proto#L2869

I also reordered the fields - it's breaking I know, but before this point no plugin has ever generated a declared resource.

Reply:
>  it's breaking I know, but before this point no plugin has ever generated a declared resource so it should be fine.

[0.4.2 went out today][042], do we still feel this is okay? It's unlikely a plugin would be developed that uses DeclaredResources yet that isn't developed by us, but we should still tread lightly 

[042]: https://github.com/hashicorp/waypoint/releases/tag/v0.4.2

Reply:
We haven't backported these protobufs, so we should be good from a release compatibility standpoint.

Reply:
I saw a few incompatibility error messages while testing with these changes, and while i'm pretty sure they're the result of something else, I'm now convinced that getting a nice number order here isn't worth the risk of breaking anything, so I reverted the order change.